homeassistant/components/lgthinq/__init__.py

```diff
@@ -0,0 +1,331 @@
+from homeassistant.util.json import JsonValueType
+
+from .const import (
+    CONF_CONNECT_CLIENT_ID,
+    DEFAULT_COUNTRY,
+    DOMAIN,
+    SERVICE_ATTR_DEVICE_INFO,
+    SERVICE_ATTR_RESULT,
+    SERVICE_ATTR_VALUE,
+)
+from .device import LGDevice, async_setup_lg_device
+from .thinq import ThinQ
+
+type ThinqConfigEntry = ConfigEntry[ThinqData]
+
+
+@dataclass(kw_only=True)
+class ThinqData:
+    """A class that holds runtime data."""
+
+    device_map: dict[str, LGDevice] = field(default_factory=dict)
+
+
+PLATFORMS = [
+    Platform.SWITCH,
+]
+
+SERVICE_GET_DEVICE_PROFILE_SCHEMA = vol.Schema(
+    {vol.Required(ATTR_DEVICE_ID): cv.string}
+)
+SERVICE_GET_DEVICE_STATUS_SCHEMA = vol.Schema({vol.Required(ATTR_DEVICE_ID): cv.string})
+SERVICE_POST_DEVICE_STATUS_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_DEVICE_ID): cv.string,
+        vol.Required(SERVICE_ATTR_VALUE): vol.Any(cv.string, dict),
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ThinqConfigEntry) -> bool:
+    """Set up an entry."""
+    _LOGGER.warning("Starting set up entry")
+
+    # Validate entry data.
+    client_id = entry.data.get(CONF_CONNECT_CLIENT_ID)
+    if not isinstance(client_id, str):
+        raise ConfigEntryError(f"Invalid client id: {client_id}")
+
+    access_token = entry.data.get(CONF_ACCESS_TOKEN)
+    if not isinstance(access_token, str):
+        raise ConfigEntryAuthFailed(f"Invalid PAT: {access_token}")
+
+    # Initialize runtime data.
+    entry.runtime_data = ThinqData()
+
+    thinq = ThinQ(
+        client_session=async_get_clientsession(hass),
+        country_code=entry.data.get(CONF_COUNTRY, DEFAULT_COUNTRY),
+        client_id=client_id,
+        access_token=access_token,
+    )
+    device_registry = dr.async_get(hass)
+
+    # Setup and register devices.
+    await async_setup_devices(hass, thinq, device_registry, entry)
+
+    # Set up all platforms for this device/entry.
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    # Set up custom services.
+    async_setup_hass_services(hass, entry)
+
+    # Clean up devices they are no longer in use.
+    async_cleanup_device_registry(
+        entry.runtime_data.device_map.values(), device_registry, entry.entry_id
+    )
+
+    return True
+
+
+async def async_setup_devices(
+    hass: HomeAssistant,
+    thinq: ThinQ,
+    device_registry: dr.DeviceRegistry,
+    entry: ThinqConfigEntry,
+) -> None:
+    """Set up and register devices."""
+    _LOGGER.warning("Starting set up devices")
+
+    entry.runtime_data.device_map.clear()
+
+    # Get a device list from the server.
+    response: ThinQApiResponse = await thinq.async_get_device_list()
+    if not ThinQ.is_success(response):
+        raise ConfigEntryError(
+            response.error_message,
+            translation_domain=DOMAIN,
+            translation_key=response.error_code,
+        )
+
+    device_list = response.body
```

Comment:
body? What is that?

Reply:
It is json result object (dictonary) for device list.